common/resource/resourceImpl.go

```diff
@@ -241,6 +247,13 @@ func New(
 		return nil, err
 	}
 
+	var isolationGroupState isolationgroup.State
+	if params.IsolationGroupState != nil {
+		isolationGroupState = params.IsolationGroupState.
+			ProvideDomainCache(domainCache).
```

Comment:
I realized that we cannot access `domainCache` and `configStoreManager` inside `cmd/server` package.
How about moving the initialization logic here? 
```
var isolationGroupState isolationgroup.State
if params.IsolationGroupState != nil {
    isolationGroupState = params.IsolationGroupState
} else {
    isolationGroupState = isolationgroup.NewDefaultIsolationGroupState(...)
}
```

Reply:
I think if you're passing in one in the params, you'll still need for it to have the Domain-cache, so I don't think that it's enough to just instantiate here. 

However, instead of passing in the module like `params.IsolationGroupState` I could pass in a factory function

```
NewDefaultIsolationGroupStateFac() func (
	logger log.Logger,
	config config.IsolationGroups,
	done chan struct{},
        domainCache domainCache,
) State {
  return func NewDefaultIsolationGroupStateWatcher( logger log.Logger,	config config.IsolationGroups, done chantruct{}, dc domainCache, cfgmgr configManager ) State { ... }
}
```
and then still instantiate that inside `resourceImpl`